helm/secrets-provider/templates/secrets-provider.yaml

```diff
@@ -65,9 +59,24 @@ spec:
         - name: CONJUR_AUTHN_LOGIN
           value:  {{ .Values.environment.conjur.authnLogin | quote }}
 
+        - name: SECRETS_DESTINATION
+          value: k8s_secrets
+
+        # Enables the support of multiple Kubernetes applications
+        - name: CONTAINER_MODE
+          value: application
+
         - name: K8S_SECRETS
           value: {{ .Values.environment.k8sSecrets | join "," }}
 
+        - name: TIMEOUT
+          value: {{- .Values.environment.conjur.timeout }}
```

Comment:
Since they are not mandatory, and do not have default values, what will be the value of this if user does not supply it? (which he doesn't even know he can...)

Reply:
this was updated to 
```

        {{- if .Values.environment.conjur.retryInterval }}
        - name: RETRY_INTERVAL
          value: {{- .Values.environment.conjur.retryInterval }}
        {{- end }}

        {{- if .Values.environment.conjur.retryCountLimit }}
        - name: RETRY_COUNT_LIMIT
          value: {{ .Values.environment.conjur.retryCountLimit }}
        {{- end }}

```

Reply:
why are you putting these "ifs" there ? the parameters are always in Values, arent they ?

Reply:
no they are optional. If the user doesn't supply them we take defaults that are defined in code

Reply:
I think the better way it always to provide these parameters from helm chart. If the user does not want to provide them he will not put them in custom-values.yaml and defaults from values.yaml will be used